### PR TITLE
Update src/secp256k1 to same as current bitcoin master and patch configure.ac

### PR DESCRIPTION
**Pull request by hans-schmidt**
Bitcoin last changed secp256k1 at commit 5803f5f5 on 10-28-2020

Bitcoin's configure.ac changes for secp256k1 ac_configure_args since the Ravencoin fork as of Bitcoin master bramch at e507acb1:
..At bf72b8a5 on 2019-08-08, bitcoin added "--enable-benchmark=no" because secp256k1 turned on benchmarks by default.
......So we do want to add that switch to our "configure.ac"
..At ca8bc423 on 2020-06-10, bitcoin dropped "--disable-jni" because they also changed bitcoin-core/secp256k1 to remove libsecp256k1's optional JNI (java native interface) in its PR#682 2020-02-10
......So we should also drop that swithc since we are updating our secp256k1 directory to Oct 2020 vintage
..At 0664f5fe on 2020-10-12 bitcoin added the schnorr switches "--enable-module-schnorrsig --enable-experimental" which we don't want.

### CODE REVIEW

**Review by fdoving — APPROVED** [review]
src/secp256k1 is identical to bitcoin master at 8f80092d78f758fdb2e79e2a832a0c7a26fa2da1

The only change outside that folder is to configure.ac

ACK aad7d0583fc9b5c7bb94dd9e0c7476b3ba93a7c1

File: src/secp256k1/contrib/travis.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+then
+    export CC="gcc-9"
+fi
+
+./configure \
+    --enable-experimental="$EXPERIMENTAL" \
+    --with-test-override-wide-multiply="$WIDEMUL" --with-bignum="$BIGNUM" --with-asm="$ASM" \
+    --enable-ecmult-static-precomputation="$STATICPRECOMPUTATION" --with-ecmult-gen-precision="$ECMULTGENPRECISION" \
+    --enable-module-ecdh="$ECDH" --enable-module-recovery="$RECOVERY" \
+    --enable-module-schnorrsig="$SCHNORRSIG" \
```
Comment (HyperPeek) [inline src/secp256k1/contrib/travis.sh:20]: Another schnorrsig enable

File: src/secp256k1/contrib/travis.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        $EXEC ./bench_recover >> bench.log 2>&1
+    fi
+    if [ "$ECDH" = "yes" ]
+    then
+        $EXEC ./bench_ecdh >> bench.log 2>&1
+    fi
+    if [ "$SCHNORRSIG" = "yes" ]
+    then
+        $EXEC ./bench_schnorrsig >> bench.log 2>&1
+    fi
```
Comment (HyperPeek) [inline src/secp256k1/contrib/travis.sh:63]: And a benchmark for Schnorsig

**Review by HyperPeek — COMMENTED** [review]
There are significant improvements in this code compared to the previous implementation:
- speed-up scalar mult functions
- using dynamic scratch memory for grid against memory-peek attack vectors
- improvement of sensitive code against timing attacks
- totally rewritten structure, much clearer
We definitely want these changes -- comparisson to bitcoin code is still pending, once finished I will approve

**Review by HyperPeek — APPROVED** [review]
src/secp256k1 confirmed to be the same as bitcoin/bitcoin as of 28.10.2020 5803f5f5f6030e69b46a46f0511b8173bf89de0d until now. Approved aad7d0583fc9b5c7bb94dd9e0c7476b3ba93a7c1

File: src/secp256k1/.travis.yml [hunk 3]
```diff
@@ -1,69 +1,108 @@
...
-    - FIELD=64bit     ENDOMORPHISM=yes
-    - FIELD=64bit     ENDOMORPHISM=yes  ECDH=yes EXPERIMENTAL=yes
-    - FIELD=64bit                       ASM=x86_64
-    - FIELD=64bit     ENDOMORPHISM=yes  ASM=x86_64
-    - FIELD=32bit     ENDOMORPHISM=yes
+    - WIDEMUL=int64   RECOVERY=yes
+    - WIDEMUL=int64   ECDH=yes  EXPERIMENTAL=yes SCHNORRSIG=yes
+    - WIDEMUL=int128
+    - WIDEMUL=int128  RECOVERY=yes EXPERIMENTAL=yes SCHNORRSIG=yes
+    - WIDEMUL=int128  ECDH=yes EXPERIMENTAL=yes SCHNORRSIG=yes
```
[the remark this answers is not in the document]
Comment (fdoving) [reply src/secp256k1/.travis.yml:26]: We don't use the .travis.yml for anything.

**Pull request merged**